plaso/storage/redis/merge_reader.py

```diff
@@ -0,0 +1,292 @@
+# -*- coding: utf-8 -*-
+"""Redis merge reader."""
+from __future__ import unicode_literals
+
+import codecs
+
+from plaso.containers import event_sources
+from plaso.containers import events
+from plaso.containers import reports
+from plaso.containers import tasks
+from plaso.containers import warnings
+from plaso.lib import definitions
+from plaso.storage import identifiers
+from plaso.storage import interface
+from plaso.storage import logger
+from plaso.storage.redis import redis_store
+
+
+class RedisMergeReader(interface.StorageMergeReader):
+  """Redis store reader for merging."""
+
+  _CONTAINER_TYPE_ANALYSIS_REPORT = reports.AnalysisReport.CONTAINER_TYPE
+  _CONTAINER_TYPE_EVENT = events.EventObject.CONTAINER_TYPE
+  _CONTAINER_TYPE_EVENT_DATA = events.EventData.CONTAINER_TYPE
+  _CONTAINER_TYPE_EVENT_SOURCE = event_sources.EventSource.CONTAINER_TYPE
+  _CONTAINER_TYPE_EVENT_TAG = events.EventTag.CONTAINER_TYPE
+  _CONTAINER_TYPE_EXTRACTION_ERROR = warnings.ExtractionError.CONTAINER_TYPE
+  _CONTAINER_TYPE_TASK_COMPLETION = tasks.TaskCompletion.CONTAINER_TYPE
+  _CONTAINER_TYPE_TASK_START = tasks.TaskStart.CONTAINER_TYPE
+
+  # Some container types reference other container types, such as event
+  # referencing event_data. Container types in this tuple must be ordered after
+  # all the container types they reference.
+  _CONTAINER_TYPES = (
+      _CONTAINER_TYPE_EVENT_SOURCE,
+      _CONTAINER_TYPE_EVENT_DATA,
+      _CONTAINER_TYPE_EVENT,
+      _CONTAINER_TYPE_EVENT_TAG,
+      _CONTAINER_TYPE_EXTRACTION_ERROR,
+      _CONTAINER_TYPE_ANALYSIS_REPORT)
+
+  _ADD_CONTAINER_TYPE_METHODS = {
+      _CONTAINER_TYPE_ANALYSIS_REPORT: '_AddAnalysisReport',
+      _CONTAINER_TYPE_EVENT: '_AddEvent',
+      _CONTAINER_TYPE_EVENT_DATA: '_AddEventData',
+      _CONTAINER_TYPE_EVENT_SOURCE: '_AddEventSource',
+      _CONTAINER_TYPE_EVENT_TAG: '_AddEventTag',
+      _CONTAINER_TYPE_EXTRACTION_ERROR: '_AddError',
+  }
+
+  def __init__(self, storage_writer, task, redis_client=None):
+    """Initializes a redis storage merge reader.
+
+    Args:
+      storage_writer (StorageWriter): storage writer.
+      task (Task): the task whose store is being merged.
+      redis_client (Optional[Redis]): redis client to query. If specified, no
+          new client will be created.
+
+    Raises:
+      RuntimeError: if an add container method is missing.
+    """
+    super(RedisMergeReader, self).__init__(storage_writer)
+    self._active_container_type = None
+    self._container_types = []
+    self._active_cursor = None
+    self._add_active_container_method = None
+    self._store = redis_store.RedisStore(
+        definitions.STORAGE_TYPE_TASK,
+        session_identifier=task.session_identifier,
+        task_identifier=task.identifier)
+    self._store.Open(redis_client=redis_client)
```

Comment:
does this raise if the open fails?

Reply:
Should we catch this and raise RuntimeError ?

Reply:
I don't believe so - this is a very unlikely situation, and I think it would just complicate the stacktrace further.